### ISSUE TITLE
Missing in props validation false positive with stateless function component.

### ISSUE THREAD

**Issue by rockneurotiko** [yannickcr/eslint-plugin-react#258]
I'm getting a prop validation that I think it's a falso positive, but I'm really new in React, so maybe I'm doing something wrong.

Here are a reduced version that fail using a stateless function component:
``` js
import React, {PropTypes} from "react";
const Test = props => {
    return <div id={props.id} onSelect={props.onSelectClick}></div>;
};

Test.propTypes = {
    id: PropTypes.string.isRequired,
    onSelectClick: PropTypes.func.isRequired
};
```

However, using a class don't show the false positive:

```js
import React, {Component, PropTypes} from "react";

class Test extends Component {
    render() {
        return <div id={this.props.id} onSelect={this.props.onSelectClick}></div>;
    }
}

Test.propTypes = {
    id: PropTypes.string.isRequired,
    onSelectClick: PropTypes.func.isRequired
};
```

edit: Using version `3.6.2`

**Comment by yannickcr** [comment]
I think this was fixed with #256 (I made some fine tuning to the stateless component detection).

Update eslint-plugin-react to `3.6.3` and reopen this issue if it's not fixed for you.

**Comment by teosz** [comment]
It happens for me :

```
import React from 'react';

const requireAuth = () => {};
export default (props) => {
  const { dispatch } = props;
  const checkAuth = requireAuth.bind({ dispatch });
  return (
    <div autht={checkAuth} />
  );
};
``` 
version `3.15.0`

**Comment by AlicanC** [comment]
@teosz, where is your `propTypes`? Your code should look like this:

```javascript
import React from 'react';

export default function StatelessComponent(props) {
  const {
    // ...
  } = props;
  
  return (
    // ...
  );
}

StatelessComponent.propTypes = {
  // ...
};
```

Note: I don't use arrow functions because they are anonymous and will probably obfuscate some of the errors I get.

**Comment by 0x80** [comment]
I'm also getting false positives. The difference with your code is that my propTypes are always defined __above__ the function. This is more readable since the interface definitions is available right before the render function. Please support this syntax.

**Comment by yannickcr** [comment]
@0x80 Can you create a new issue with 

* an example of the code that do not work
* your eslint and eslint-plugin-react versions
* your .eslintrc config file

Thanks.

**Comment by 0x80** [comment]
@yannickcr ok see #472

**Comment by HienNguyenAsnet** [comment]
I got this error in eslint-plugin-react  `7.18.0`

```
const Auth = props => {
  return <>{props.children}</>
}

Auth.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]),
}

Auth.defaultProps = {
  children: null,
}
```
